server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityAction.java

```diff
@@ -280,6 +280,21 @@ private static Long parseDateTimeAsLong(@Nullable String dateAsString) {
     return date.getTime();
   }
 
+  @CheckForNull
+  private static Long parseFromDate(@Nullable String dateAsString) {
+    if (dateAsString == null) {
+      return null;
+    }
+
+    Date date = parseDateTimeQuietly(dateAsString);
+    if (date == null) {
+      date = parseDateQuietly(dateAsString);
+      checkRequest(date != null, "Date '%s' cannot be parsed as either a date or date+time", dateAsString);
+    }
```

Comment:
I agree, I'll refactor later on (when handling QP Changelog). I'll add 2 methods in DateUtils: `parseEndingDate(String)` and `parseStartingDate(String)`